app/src/main/java/com/edu/uni/augsburg/uniatron/MainActivity.java

```diff
@@ -44,6 +52,45 @@ protected void onCreate(Bundle savedInstanceState) {
         mTextMessage = (TextView) findViewById(R.id.message);
         BottomNavigationView navigation = (BottomNavigationView) findViewById(R.id.navigation);
         navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
+
+
+        // start the service when the app is first opened after installation
+        // rebooting will automatically start the service via BroadcastReceiverOnStateChanged class
+        // force closing via OS will automatically start the service because it's sticky
+        checkFirstRun();
+        }
+
+
+    private void checkFirstRun() {
+
+        // Get current version code
+        int currentVersionCode = BuildConfig.VERSION_CODE;
+
+        // Get saved version code
+        SharedPreferences prefs = getSharedPreferences(PREFS_NAME, MODE_PRIVATE);
```

Comment:
Move every SharedPreferences stuff to an own class. There we can collect all the preferences things, which will become more over time. Then you just need to call the class XY.isFirstRun() to verify the first execution of the app.